core/controllers/reader.py

```diff
@@ -310,11 +310,16 @@ def post(self, exploration_id):
         old_state_name = self.payload.get('old_state_name')
         # The reader's answer.
         answer = self.payload.get('answer')
+        # The answer handler (submit, click, etc.)
+        handler_name = self.payload.get('handler')
```

Comment:
Follow-up: I misunderstood the comment. Yes, I'm not sure why these lines were here. They shouldn't be; I've removed them. I also did a quick check to make sure handler_name isn't anywhere it shouldn't be.